docs/route/route.md

```diff
@@ -74,9 +76,12 @@ interface RouteObject {
   loader?: LoaderFunction;
   action?: ActionFunction;
   element?: React.ReactNode | null;
+  Component?: React.FunctionComponent | null;
```

Comment:
Is it a deliberate choice to make class components a type error?

Reply:
Hm, sort of?  We should check with Michael.  I chose `FunctionComponent` because you can't access any of the v6 hooks otherwise, but I guess that doesn't mean you can't use a top-level class component?  Can they render functional components as children?  Here's a relevant thread on the v6 preference for functional components: https://github.com/remix-run/react-router/issues/8146#issuecomment-947860640

Reply:
As far as I know, class components can render any kind of component (so both class & function components) as a child.

```suggestion
  Component?: React.ComponentType | null;
```

Reply:
In hindsight, I should've just looked at what Remix uses for the route module `default` export - which is indeed `React.ComponentType` :)

Reply:
Updated in [2c9f13d](https://github.com/remix-run/react-router/pull/10045/commits/2c9f13d810b430985bb702958e922e4cb295f702)